kotlinpoet/src/test/java/com/squareup/kotlinpoet/DelegatedConstructorCallTest.kt

```diff
@@ -0,0 +1,156 @@
+    builder.superclass(ClassName("testpackage", "TestSuper"))
+    assertThat(builder.build().toString()).isEqualTo(
+      """
+        |public external class Test : testpackage.TestSuper
+        |""".trimMargin()
+    )
+  }
+
+  @Test
+  fun defaultNotPresentInExpectClass() {
+    val builder = TypeSpec.classBuilder("Test")
+    builder.addModifiers(KModifier.EXPECT)
+    builder.superclass(ClassName("testpackage", "TestSuper"))
+    assertThat(builder.build().toString()).isEqualTo(
+      """
+        |public expect class Test : testpackage.TestSuper
+        |""".trimMargin()
+    )
+  }
+
+  @Test
+  fun defaultNotPresentInExpectObject() {
+    val builder = TypeSpec.objectBuilder("Test")
+    builder.addModifiers(KModifier.EXPECT)
+    builder.superclass(ClassName("testpackage", "TestSuper"))
+    assertThat(builder.build().toString()).isEqualTo(
+      """
+        |public expect object Test : testpackage.TestSuper
+        |""".trimMargin()
+    )
+  }
+
+  @Test
+  fun defaultNotPresentInExternalObject() {
+    val builder = TypeSpec.objectBuilder("Test")
+    builder.addModifiers(KModifier.EXTERNAL)
+    builder.superclass(ClassName("testpackage", "TestSuper"))
+    assertThat(builder.build().toString()).isEqualTo(
+      """
+        |public external object Test : testpackage.TestSuper
+        |""".trimMargin()
+    )
+  }
+
+  @Test
+  fun allowedInClass() {
+    val builder = TypeSpec.classBuilder("Test")
+    builder.superclass(ClassName("testpackage", "TestSuper"))
+    builder.addSuperclassConstructorParameter("anything")
+    assertThat(builder.build().toString()).isEqualTo(
+      """
+        |public class Test : testpackage.TestSuper(anything)
+        |""".trimMargin()
+    )
+  }
+
+  @Test
+  fun allowedInObject() {
+    val builder = TypeSpec.objectBuilder("Test")
+    builder.superclass(ClassName("testpackage", "TestSuper"))
+    builder.addSuperclassConstructorParameter("anything")
+    assertThat(builder.build().toString()).isEqualTo(
+      """
+        |public object Test : testpackage.TestSuper(anything)
+        |""".trimMargin()
+    )
+  }
+
+  @Test
+  fun allowedInClassSecondary() {
+    val builder = TypeSpec.classBuilder("Test")
+    val primaryConstructorBuilder = FunSpec.constructorBuilder()
+    val primaryConstructor = primaryConstructorBuilder.build()
+    builder.primaryConstructor(primaryConstructor)
+    val secondaryConstructorBuilder = FunSpec.constructorBuilder()
+    secondaryConstructorBuilder.addParameter(ParameterSpec("foo", ClassName("kotlin", "String")))
+    secondaryConstructorBuilder.callThisConstructor()
+    builder.addFunction(secondaryConstructorBuilder.build())
+    assertThat(builder.build().toString()).isEqualTo(
+      """
+        |public class Test {
+        |  public constructor(foo: kotlin.String) : this()
+        |}
+        |""".trimMargin()
+    )
+  }
+
+  @Test
+  fun notAllowedInExternalClass() {
+    val builder = TypeSpec.classBuilder("Test")
```

Comment:
nit: the code could look a bit less verbose if you used method chaining:
```kotlin
val builder = TypeSpec.classBuilder("Test")
  .addModifiers(KModifier.EXTERNAL)
  .superclass(ClassName("testpackage", "TestSuper"))
  .addSuperclassConstructorParameter("anything")
```
As is is fine though, so not critical to fix.

Reply:
@Egorand Done. CLA also signed.